bin/build_sdb.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python
+'''
+Builds Sample Databases (.sdb files)
+Use "python3 build_sdb.py -h" for help
+'''
+from __future__ import absolute_import, division, print_function
+
+# Make sure we can import stuff from util/
+# This script needs to be run from the root of the DeepSpeech repository
+import os
+import sys
+sys.path.insert(1, os.path.join(sys.path[0], '..'))
+
+import argparse
+import progressbar
+
+from util.downloader import SIMPLE_BAR
+from util.helpers import parse_file_size
+from util.audio import convert_samples, AUDIO_TYPE_OPUS
+from util.collections import samples_from_files, DirectSDBWriter, SortingSDBWriter
+
+
+def add_samples(sample_sink):
+    samples = samples_from_files(CLI_ARGS.sources)
+    num_workers = os.cpu_count() * CLI_ARGS.load_factor if CLI_ARGS.load_factor > 0 else 1
+    bar = progressbar.ProgressBar(max_value=len(samples), widgets=SIMPLE_BAR)
+    for sample in bar(convert_samples(samples, audio_type=AUDIO_TYPE_OPUS, processes=num_workers)):
+        sample_sink.add(sample)
+
+
+def build_sdb():
+    if CLI_ARGS.sort:
+        with SortingSDBWriter(CLI_ARGS.target,
+                              tmp_sdb_filename=CLI_ARGS.sort_tmp_file,
+                              cache_size=parse_file_size(CLI_ARGS.sort_cache_size)) as sdb_writer:
+            add_samples(sdb_writer)
+    else:
+        with DirectSDBWriter(CLI_ARGS.target) as sdb_writer:
+            add_samples(sdb_writer)
+
+
+def handle_args():
+    parser = argparse.ArgumentParser(description='Tool for building Sample Databases (.sdb files) '
+                                                 'from DeepSpeech CSV files')
+    parser.add_argument('--load_factor', type=int, default=1,
+                        help='CPU-multiplier for the number of parallel workers - 0 for 1 worker')
+    parser.add_argument('--sort', action='store_true', help='Force sample sorting by durations '
```

Comment:
SDBs have to be pre-ordered, as they will be streamed in feeding. Efficient ordering of a huge series of BLOBs is only possible with "external sorting".

Reply:
> SDBs have to be pre-ordered, as they will be streamed in feeding. Efficient ordering of a huge series of BLOBs is only possible with "external sorting".

Ok, I guess it's confusing then. Does it make sense to disable ordering ? "Force" suggests it's not the default, but your reply suggests it makes no sense to not sort.

Reply:
It makes sense in general, the only reason why we always use the data in sorted order is because we were too lazy to implement SortaGrad. First it was awkward to do with the feeding queues, then I didn't bother to do with on the tf.data transition, and now we are encoding the notion that ordering does not change during a training run here in the SDB format as well. This is not the norm, in fact the standard procedure is to shuffle examples or batches on every epoch.

Reply:
Should we drop sorting then?

Reply:
I was thinking more drop the flag and always sort the files.

Reply:
The flag will now get dropped as part of the decision to remove `SortingSDBWriter`.